cpp/src/DataStorm/Node.cpp

```diff
@@ -93,13 +94,13 @@ Node::Node(optional<string_view> configFile, function<void(function<void()> call
 {
 }
 
-Node::Node(Ice::CommunicatorPtr communicator, function<void(function<void()> call)> customExecutor)
-    : Node(std::move(communicator), std::move(customExecutor), false)
+Node::Node(const CommunicatorPtr& communicator, function<void(function<void()> call)> customExecutor)
+    : Node(communicator, std::move(customExecutor), false)
 {
 }
 
 Node::Node(
-    Ice::CommunicatorPtr communicator,
+    const CommunicatorPtr& communicator,
```

Comment:
I assume _instance below keeps a refcount on communicator, so it should be by value. You may want to fix the DataStormI::Instance ctor - you actually already fixed it!